api/handlers/ipfs/ipfs_peer/ipfs_peer_handler.go

```diff
@@ -154,3 +175,168 @@ func ValidatePeerExist(c *fiber.Ctx) error {
 
 	return c.Next()
 }
+
+// Stats returns a websocket that emits peers, bw ,pin and files stats
+func Stats(c *websocket.Conn) {
+	defer c.Close()
+
+	name := c.Params("name")
+	peer := &ipfsv1alpha1.Peer{}
+	nameSpacedName := types.NamespacedName{
+		Namespace: c.Locals("namespace").(string),
+		Name:      name,
+	}
+	err := k8sClient.Get(context.Background(), nameSpacedName, peer)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			c.WriteJSON(fiber.Map{
+				"error": fmt.Sprintf("peer by name %s doesn't exist", name),
+			})
+			return
+		}
+		c.WriteJSON(fiber.Map{
+			"error": err.Error(),
+		})
+		return
+	}
+
+	if !peer.Spec.API {
+		c.WriteJSON(fiber.Map{
+			"error": "peer api is not enabled",
+		})
+		return
+	}
+
+	for {
+		jobs := make(chan statCalls, 4) //buffered job channel
+		results := make(chan result, 4) //buffered results channel
+
+		//loop over a worker that accepts jobs receive-channel and results send-channel
+		for i := 0; i < 4; i++ { //the total execution time decreases with the increase of this loop number
+			go worker(jobs, results)
+		}
```

Comment:
the number of iterations here can vary from the length of the jobs channel, for example i can choose to create only 2 go-routine to run all the jobs.
another example,
if i have 100 jobs to do, it would be a bad idea to create 100 go-routine , but i can choose two create only 5 go routine to  handle the 100 job.